homeassistant/components/automation/state.py

```diff
@@ -75,7 +75,7 @@ def call_action():
                             "for": time_delta if not time_delta else period[entity],
                         }
                     },
-                    context=to_s.context,
+                    context=to_s.context if to_s else None,
```

Comment:
This is not correct. There is a context, it just needs to be extracted from the state changed event.

The correct solution here is that we listen to `EVENT_STATE_CHANGED` event directly and process it accordingly.

Reply:
Ok, I'll try that (I saw that context was an optional parameter and tried the quick fix).

> The correct solution here is that we listen to EVENT_STATE_CHANGED event directly and process it accordingly.

Then you're suggesting to basically copy the state comparison logic from `async_track_state_change` here in automation, right? To re-implement the`state_change_listener` for this trigger. Is that it?

Reply:
Yes.